docs/s3.md

```diff
@@ -0,0 +1,105 @@
+# S3 compatible dataset API
+
+Palantir Foundry released a beta for a S3 compatible dataset API. Which lets you use the aws cli, boto3 and other s3 compatible libraries.
+As the authentication via a JWT does not work directly, but needs another API call inbetween, we created these methods and the cli to get you easily started with the API.
+
+
+## CLI
+
+### Init
+
+For the AWS cli, we needed to create a profile in your config, which dynamically hands over the S3 credentials to the cli.
+
+To create this profile in your config run:
+
+```zsh
+fdt s3 init
+```
+
+This will do the following things:
+
+1. Read your current aws config file, if it does not exist, it will create one
+2. Add the foundry profile programmatically, which uses the `credential_config` option to call the `fdt s3 auth` cli.
+3. Show a diff to the previous config and create a backup of the old config, if a change was made.
+
+After running the command, please check the diff and your aws config file,
+you can always recover the config via the backup file, the path to the backup will be printed, and is in the same directory as your aws config.
+
+### Auth
+
+```zsh
+fdt s3 auth
+```
+
+The auth command gets used by the configuration profile as mentioned above, it adheres to the output schema defined at https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html
+
+
+## Methods you can use in python
+
+- [FoundryRestClient.get_s3_client](#foundry_dev_tools.foundry_api_client.FoundryRestClient.get_s3_client)
+- [FoundryRestClient.get_s3_resource](#foundry_dev_tools.foundry_api_client.FoundryRestClient.get_s3_resource)
```

Comment:
Let’s allow user to pass config or additional kwargs, for example to set retry settings https://stackoverflow.com/a/48568320